modules/oe_editorial_corporate_workflow/oe_editorial_corporate_workflow.module

```diff
@@ -56,20 +61,162 @@ function oe_editorial_corporate_workflow_menu_local_tasks_alter(&$data): void {
+
+  $form_options = $options;
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * For the content moderation block form to make sure next state is selected
+ * in the widget.
+ */
+function oe_editorial_corporate_workflow_form_content_moderation_entity_moderation_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  if (empty($form['new_state']['#options'])) {
+    return;
+  }
+
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = \Drupal::routeMatch()->getParameter('node');
+  /** @var \Drupal\content_moderation\ModerationInformationInterface $moderation_info */
+  $moderation_info = \Drupal::service('content_moderation.moderation_information');
+  /** @var \Drupal\workflows\WorkflowInterface $workflow */
+  $workflow = $moderation_info->getWorkflowForEntity($node);
+
+  if (count($form['new_state']['#options']) > 1) {
+    $upcoming_transitions = $workflow->getTypePlugin()->getTransitionsForState($node->moderation_state->value);
+    $transition = end($upcoming_transitions);
+    if (in_array($transition->to()->id(), array_keys($form['new_state']['#options']))) {
+      $form['new_state']['#default_value'] = $transition->to()->id();
+    }
+  }
+
+  // Define our submit callback as the first to execute.
+  array_unshift($form['#submit'], '_oe_editorial_corporate_workflow_moderation_submit');
+}
+
+/**
+ * Submit callback to handle revisions when shortcuts are used on the moderation
+ * form.
+ *
+ * We need to make sure that intermediary states and revisions are saved so that
+ * related events are executed at every step.
+ *
+ * @param array $form
+ *   The form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function _oe_editorial_corporate_workflow_moderation_submit(array &$form, FormStateInterface &$form_state): void {
+  /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+  $entity = $form_state->get('entity');
+  /** @var \Drupal\content_moderation\ModerationInformationInterface $moderation_info */
+  $moderation_info = \Drupal::service('content_moderation.moderation_information');
+  /** @var \Drupal\workflows\WorkflowInterface $workflow */
+  $workflow = $moderation_info->getWorkflowForEntity($entity);
+  /** @var \Drupal\workflows\WorkflowTypeInterface $workflow_plugin */
+  $workflow_plugin = $workflow->getTypePlugin();
+
+  // We start the saving from the current state of the entity and we proceed
+  // until the target state that was selected from the moderation form.
+  $current_state = $entity->get('moderation_state')->value;
+  $target_state = $form_state->getValue('new_state');
+
+  _oe_editorial_corporate_workflow_do_transition_revisions($current_state, $target_state, $workflow_plugin, $entity, $form_state);
+}
+
+/**
+ * Helper function to do state transitions and revisions to the target state.
+ *
+ * States between the current and target should not be skipped with their
+ * revisions. This function ensures that the workflow steps are simulated
```

Comment:
Not the workflow steps are simulated, but the transitions are executed as they would be manually saved with an associated revision save per transition.